From cc5f351bcbab79c5131e108cf5605f673abeb8c6 Mon Sep 17 00:00:00 2001
From: Javier Maestro <jjmaestro@ieee.org>
Date: Sun, 3 May 2026 00:51:23 +0000
Subject: [PATCH] patch: use POSIX short flags to decompress the Package index

Short flags are POSIX-standard and accepted by every implementation
(GNU, busybox, BSD, Alpine). Thus, this is a portability improvement
that allows decompression to work on hosts or sandboxes with non-GNU
tools (e.g. busybox-static).
---
 apt/private/apt_deb_repository.bzl | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/apt/private/apt_deb_repository.bzl b/apt/private/apt_deb_repository.bzl
index 95657f5..8564cf2 100644
--- a/apt/private/apt_deb_repository.bzl
+++ b/apt/private/apt_deb_repository.bzl
@@ -15,15 +15,16 @@ def _get_auth(ctx, urls):
 def _fetch_package_index(rctx, urls, dist, comp, arch, integrity):
     target_triple = "{dist}/{comp}/{arch}".format(dist = dist, comp = comp, arch = arch)
 
+    # Using POSIX short flags because these are accepted by every implementation, including busybox
     # See https://linux.die.net/man/1/xz , https://linux.die.net/man/1/gzip , and https://linux.die.net/man/1/bzip2
-    #  --keep       -> keep the original file (Bazel might be still committing the output to the cache)
-    #  --force      -> overwrite the output if it exists
-    #  --decompress -> decompress
+    #  -k -> keep the original file (Bazel might be still committing the output to the cache)
+    #  -f -> overwrite the output if it exists
+    #  -d -> decompress
     # Order of these matter, we want to try the one that is most likely first.
     supported_extensions = [
-        (".xz", ["xz", "--decompress", "--keep", "--force"]),
-        (".gz", ["gzip", "--decompress", "--keep", "--force"]),
-        (".bz2", ["bzip2", "--decompress", "--keep", "--force"]),
+        (".xz", ["xz", "-d", "-k", "-f"]),
+        (".gz", ["gzip", "-d", "-k", "-f"]),
+        (".bz2", ["bzip2", "-d", "-k", "-f"]),
         ("", ["true"]),
     ]
 
